File: android/core/model/src/main/java/com/bolsaterminal/core/model/Broker.kt
package com.bolsaterminal.core.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class OrderSide {
    @SerialName("BUY") Buy,
    @SerialName("SELL") Sell,
}

@Serializable
enum class OrderStatus {
    @SerialName("FILLED") Filled,
    @SerialName("CANCELLED") Cancelled,
    @SerialName("REJECTED") Rejected,
}

@Serializable
enum class TradeSource {
    @SerialName("MANUAL") Manual,
    @SerialName("BOT") Bot,
}

@Serializable
data class BrokerOrder(
    val id: String,
    val symbol: String,
    val side: OrderSide,
    val quantity: Double,
    val price: Double,
    val total: Double,
    val fee: Double,
    val status: OrderStatus,
    val timestamp: Double,
    val source: TradeSource,
    val reason: String? = null,
)

@Serializable
data class BrokerPosition(
    val symbol: String,
    val quantity: Double,
    val avgCost: Double,
    val currentPrice: Double,
    val value: Double,
    val cost: Double,
    val pnl: Double,
    val pnlPct: Double,
    val stopLoss: Double? = null,
    val takeProfit: Double? = null,
    val openedAt: Double,
    val source: TradeSource,
)

@Serializable
data class BrokerAccount(
    val cash: Double,
    val initialBalance: Double,
    val positions: List<BrokerPosition>,
    val orders: List<BrokerOrder>,
    val totalEquity: Double,
    val totalCost: Double,
    val totalPnL: Double,
    val totalPnLPct: Double,
    val totalFeesPaid: Double,
    val tradeCount: Int,
    val winCount: Int,
    val lossCount: Int,
    val winRate: Double,
)

File: macos/BolsaTerminal/Core/Models/Broker.swift
import Foundation

enum OrderSide: String, Codable {
    case buy = "BUY"
    case sell = "SELL"
}

enum OrderStatus: String, Codable {
    case filled = "FILLED"
    case cancelled = "CANCELLED"
    case rejected = "REJECTED"
}

enum TradeSource: String, Codable {
    case manual = "MANUAL"
    case bot = "BOT"
}

struct BrokerOrder: Codable, Hashable, Identifiable {
    let id: String
    let symbol: String
    let side: OrderSide
    let quantity: Double
    let price: Double
    let total: Double
    let fee: Double
    let status: OrderStatus
    let timestamp: Double
    let source: TradeSource
    let reason: String?
}

struct BrokerPosition: Codable, Hashable, Identifiable {
    var id: String { symbol }
    let symbol: String
    let quantity: Double
    let avgCost: Double
    let currentPrice: Double
    let value: Double
    let cost: Double
    let pnl: Double
    let pnlPct: Double
    let stopLoss: Double?
    let takeProfit: Double?
    let openedAt: Double
    let source: TradeSource
}

struct BrokerAccount: Codable, Hashable {
    let cash: Double
    let initialBalance: Double
    let positions: [BrokerPosition]
    let orders: [BrokerOrder]
    let totalEquity: Double
    let totalCost: Double
    let totalPnL: Double
    let totalPnLPct: Double
    let totalFeesPaid: Double
    let tradeCount: Int
    let winCount: Int
    let lossCount: Int
    let winRate: Double
}
